actionpack/lib/action_dispatch/journey/gtg/transition_table.rb

```diff
@@ -41,22 +45,50 @@ def eclosure(t)
           Array(t)
         end
 
-        def move(t, a)
+        def move(t, full_string, start_index, end_index)
           return [] if t.empty?
 
-          regexps = []
-          strings = []
+          next_states = Set.new
+
+          tok = full_string.slice(start_index, end_index - start_index)
+          token_matches_default_component = DEFAULT_EXP_ANCHORED.match?(tok)
+
+          t.each { |s, previous_start|
+            # In the simple case of a "default" param regex do this fast-path
+            # and add all next states.
+            if previous_start.nil? && token_matches_default_component && states = @stdparam_states[s]
```

Comment:
In theory this means non-constrained components at the same level don't require repeated regexp matching. We could look at whether this is actually more performant than simplifying the code to just always check with the more complex logic below.